core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexIT.java

```diff
@@ -244,10 +247,15 @@ public void onFailure(Throwable e) {
         for (int i = 0; i < numDocs; i++) {
             try {
                 synchronized (indexVersionLock) {
-                    client().prepareIndex("test", "test").setSource("index_version", indexVersion.get()).get();
+                    client().prepareIndex("test", "test").setSource("index_version", indexVersion.get()).setTimeout(TimeValue.timeValueSeconds(10)).get();
                 }
             } catch (IndexNotFoundException inf) {
-                // fine
+                // fine index has been deleted
+            } catch (ProcessClusterEventTimeoutException ex) {
+                // fine request timed out since we where waiting for a shard and maybe index was deleted?
+            } catch (UnavailableShardsException ex) {
```

Comment:
we do if the index is not there, but there is a racing condition where the index is there when we check for that and then the shards are deleted beneath us. Then we go and wait for the shards (identically to where the index is created but the shards are not assigned yet), which results in this exception. We can do another check before failing, but I don't think it's worth it for this edge case?